### ISSUE TITLE
Support selecting columns using existing column names

### ISSUE THREAD

**Issue by unclebob418** [zio/zio-sql#28]
Currently we need to specify the column names using `as`, but some columns already have names defined e.g. for source columns. which were specified when we defined the table. These names should be used as default when "as" is not specified. Perhaps we need a new "Named" `Feature`?

```scala
object Example {
  new Sql { self =>
    import self.ColumnSet._

    val userTable = (uuid("user_id") ++ localDate("dob") ++ string("first_name") ++ string("last_name")).table("users")

    val userId :*: dob :*: fName :*: lName :*: _ = userTable.columns

    //val basicSelect = select { fName ++ lName } from userTable //todo this should compile using column names defined in the table

    // fName and lName already have column names, shouldn't have to do this
    val basicSelectWithAliases = (select {
      (fName as "first_name") ++ (lName as "last_name")
    } from userTable)
  }
}
```

**Comment by jgoday** [comment]
Could the use of implicit conversions be a valid solution ?
I mean, we can implicit convert an Expression to a Selection and then use the existing ++ operator.

```scala
def columnName[F, A, B](expr: Expr[F, A, B]): Option[String] =
    expr match {
      case Expr.Source(_, c) => Some(c.name)
      case _ => None
    }

implicit def expToSelection[F, A, B](expr: Expr[F, A, B]) =
  Selection.computedOption(expr, columnName(expr))
```

and then we can use it like the previous example
```scala
val basicSelect = select { fName ++ lName } from userTable //todo this should compile using column names defined in the table

// fName and lName already have column names, shouldn't have to do this
val basicSelectWithAliases = (select {
  (fName as "first_name") ++ (lName as "last_name")
} from userTable)

...

test("basic select") {
   assert(basicSelect)(equalTo(basicSelectWithAliases))
}

```

Representation of basicSelect is in fact =
```scala
Select(Selection(
    Cons(Computed(Source(users,Column(first_name)),Some(first_name)),
    Cons(Computed(Source(users,Column(last_name)),Some(last_name)),Empty))),
     zio.sql.Sql$ColumnSet$Cons$$anon$1@782715e4,Literal(true),List(),List(),None,None)
```

**Comment by unclebob418** [comment]
I think it will have to be an implicit conversion, but I think it needs to be a little more complicated to support things other that sources that have been named e.g. below we still have to specify a name for Arbitrary(age), and count isn't a source.

```scala
val aggregated =
      (select {
        (Arbitrary(age) as "age") ++ (Count(1) as "count")
      } from table).groupBy(age).table("age_counts")

    val age :*: count :*: _ = 
```

I've just realised this example won't actually work (types for the deconstructed columns are `Any`), but I think it should if we're going to be trying to make this as composable as possible. I'll raise a separate issue for that.

**Comment by unclebob418** [comment]
edit: ignore my last, that example is very broken for many reasons, I think your suggestion will be the best option for now - can you please raise a PR?

**Comment by jgoday** [comment]
> edit: ignore my last, that example is very broken for many reasons, I think your suggestion will be the best option for now - can you please raise a PR?

Thank you for your answer!

**Comment by unclebob418** [comment]
fixed by #34
